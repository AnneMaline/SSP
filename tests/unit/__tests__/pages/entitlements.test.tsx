import { fireEvent, screen, render } from "@testing-library/react";
import EntitlementsPage from "../../../../src/app/Entitlements/page";

describe("Entitlements Page", () => {
  it("renders correctly", () => {
    const { container } = render(<EntitlementsPage />);
    expect(container).toMatchSnapshot();
  });

  it("removes a group from the list when the delete button is clicked", () => {
    render(<EntitlementsPage />);

    // Verify initial state: Two groups are rendered
    expect(screen.getByText("Team A")).toBeInTheDocument();
    expect(screen.getByText("Team B")).toBeInTheDocument();

    // Open the dropdown for "Team A"
    fireEvent.click(screen.getByText("Team A"));

    // Click the delete button for "Team A"
    const deleteButton = screen.getByText("Delete");
    fireEvent.click(deleteButton);

    // Verify "Team A" is removed and "Team B" is still present
    expect(screen.queryByText("Team A")).not.toBeInTheDocument();
    expect(screen.getByText("Team B")).toBeInTheDocument();
  });
});
