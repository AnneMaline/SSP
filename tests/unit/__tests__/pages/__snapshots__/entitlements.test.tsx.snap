// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Entitlements Page renders correctly 1`] = `
<div>
  <div>
    <h1>
      Entitlements
    </h1>
    <ul
      class="mb-2 mt-2"
    >
      <li>
        <div
          class="border border-gray-300 mr-2 ml-2"
        >
          <div
            class="flex justify-between items-center px-4 py-2 cursor-pointer"
          >
            <span>
              Team A
            </span>
            <span>
              ↓
            </span>
          </div>
        </div>
      </li>
      <li>
        <div
          class="border border-gray-300 mr-2 ml-2"
        >
          <div
            class="flex justify-between items-center px-4 py-2 cursor-pointer"
          >
            <span>
              Team B
            </span>
            <span>
              ↓
            </span>
          </div>
        </div>
      </li>
    </ul>
  </div>
</div>
`;
